lib/mix/test/mix/tasks/compile.elixir_test.exs

```diff
@@ -236,6 +236,41 @@ defmodule Mix.Tasks.Compile.ElixirTest do
 
       assert Mix.Tasks.Compile.Elixir.run(["--elixirc-paths", "lib"])
       refute_received {:mix_shell, :info, ["Compiled lib/a.ex"]}
+
+      # Compiling just web and then just lib should recompile lib
+      # if both are modified
+      :timer.sleep(1000)
```

Comment:
I wasn't able to reproduce the bug without sleeping. Setting the mtime in the future didn't expose the bug properly. The mtime needs to be between the first time `lib/a.ex` was compiled and when `web/ab.ex` was compiled.

Any suggestions here? The only approach I can think of is to set the mtime on the manifest manually, which is a little grey boxish instead of black box.